k9mail/src/main/java/com/fsck/k9/view/MessageCryptoDisplayStatus.java

```diff
@@ -237,12 +252,15 @@ private static MessageCryptoDisplayStatus getDisplayStatusForPgpResult(CryptoRes
             }
         }
 
+        final long delta = signatureResult.getSignatureTimestamp().getTime() - messageSentDate.getTime();
```

Comment:
I don't care too much which app makes the decision.

But something that needs special consideration is a message without Date header.